backend/common/src/main/java/ai/verta/modeldb/common/reconcilers/Reconciler.java

```diff
@@ -40,7 +40,9 @@ protected Reconciler(
     this.executor = executor;
     this.deduplicate = deduplicate;
 
-    startResync();
+    if (!config.isTestReconciler()) {
+      startResync();
+    }
```

Comment:
I'm not a big fan of code that behaves differently during tests. Why do we need this? Can we make things work in a way that is the same in test or out?

Reply:
this will be starting new periodic thread to fetched deleted entity and try to delete but in the test cases we are calling directly `resync` and not need to start the periodic thread. if we will depends on it then in between tests it will difficult to identify when it will execute and how much time we will wait to finish it. other wise all other code will used the same one.